kernel/packages/shared/session/index.ts

```diff
@@ -119,6 +120,18 @@ export async function userAuthentified(): Promise<void> {
   })
 }
 
+export function authenticateWhenItsReady(providerType: ProviderType | null) {
```

Comment:
Maybe you want to check if we are already at the sign_in state?
```typescript
export function authenticateWhenItsReady(providerType: ProviderType | null) {
  const store: Store<RootState> = globalThis.globalStore
  if (loginStage === LoginStage.SIGN_IN) {
    globalThis.globalStore.dispatch(authenticate(providerType))
  }
  const unsubscribe = store.subscribe(() => {
    const loginStage = store.getState().session.loginStage
    if (loginStage === LoginStage.SIGN_IN) {
      unsubscribe()
      globalThis.globalStore.dispatch(authenticate(providerType))
    }
  })
}
```

I'm not sure about the possible flows, so this may not be necessary. But I'm asking since this may be a gotcha.

Reply:
Yeah, you're right, and I would like to ignore the message if we already signed in.

I pushed the changes, what do you think now?

Reply:
Looks good! 

Just as an extra mile, you may want to put the accesses to the store in a selector, since that is considered best practice for accessing state in redux and similar: https://github.com/decentraland/explorer/blob/master/kernel/packages/shared/session/selectors.ts

But feel free to ignore this since we are not respecting that convention in a bunch of places already hahaha.